File: jp_backend/Models/Details/DinosaurDetails.cs
using jp_backend.Database.Entities;

namespace jp_backend.Models.Details
{
    public class DinosaurDetails
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public int? DiscoveryYear { get; set; }

        // Informations
        public string? EatingPattern { get; set; }
        public string? ModeOfLocomotion { get; set; }
        public int HeightInCentimeter { get; set; }
        public int LengthInCentimeter { get; set; }
        public double WeightInKilogram { get; set; }
        public List<string> DnaString { get; set; }

        // Fremdverweise der Klasse Dinosaurier
        public PeriodDetails? Period { get; set; }
        public DinosaurClassDetails? Classification { get; set; }
        public DinosaurLocalityDetails? LocalityOfDiscovery { get; set; }

        // Thumbnail
        public Guid? ThumbnailId { get; set; }
        public string? ThumbnailDownloadUrl { get; set; }
        public FileReferenceDetails? Thumbnail { get; set; }
    }
}

File: jp_frontend/frontend/src/app/api/models/dinosaur-details.ts
/* tslint:disable */
/* eslint-disable */
import { DinosaurClassDetails } from './dinosaur-class-details';
import { DinosaurLocalityDetails } from './dinosaur-locality-details';
import { FileReferenceDetails } from './file-reference-details';
import { PeriodDetails } from './period-details';
export interface DinosaurDetails {
  classification?: DinosaurClassDetails;
  description?: null | string;
  discoveryYear?: null | number;
  dnaString?: null | Array<string>;
  eatingPattern?: null | string;
  heightInCentimeter?: number;
  id?: string;
  lengthInCentimeter?: number;
  localityOfDiscovery?: DinosaurLocalityDetails;
  modeOfLocomotion?: null | string;
  name?: null | string;
  period?: PeriodDetails;
  thumbnail?: FileReferenceDetails;
  thumbnailDownloadUrl?: null | string;
  thumbnailId?: null | string;
  versionNumber?: null | string;
  weightInKilogram?: number;
}
